app/com/linkedin/drelephant/exceptions/TonYExceptionFingerprinting.java

```diff
@@ -339,15 +342,20 @@ private boolean isExceptionFingerPrintingAlreadyDone(String appId) {
    * @param exceptionStackTrace HashCode for filterOut exception stackTrace
    * @return where this exception stackTrace is parsed and saved before
    */
-  private boolean isExceptionLogDuplicate(String exceptionStackTrace) {
+  private boolean isExceptionLogDuplicateOrSimilar(String exceptionStackTrace) {
```

Comment:
Duplicate is subset of Similar , please change the method name to isExceptionLogSimilar. 
Also why are you checking for Duplicate and similarity separately .

Reply:
I am checking first if the `exceptionIdSet` has the newly parsed `exceptionStackTrace` or not, if this comes true means there is high probability that a DUPLICATE log is parsed before. Complexity for checking string equality is O(n) and for checking string similarity is O(mn).